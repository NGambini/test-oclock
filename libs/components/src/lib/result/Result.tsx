import React from 'react';
import CountUp from 'react-countup';

import * as styles from './result.module.less';

export type ResultProps = {
  value: number;
  symbol: string;
};

export const Result = (props: ResultProps) => {
  return (
    <div data-cy="result" className={styles['result-container']}>
      <div className={styles['result-value']}>
        <CountUp end={props.value} />
        {'  ' + props.symbol}
      </div>
      <div className={styles['result-clearfix']} />
    </div>
  );
};
