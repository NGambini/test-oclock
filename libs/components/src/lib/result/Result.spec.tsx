import React from 'react';
import { render } from '@testing-library/react';

import { Result } from './Result';

describe('Result', () => {
  it('renders correctly', () => {
    const { baseElement: result } = render(<Result symbol="USD" value={50} />);
    expect(result).toMatchSnapshot();
  });
});
