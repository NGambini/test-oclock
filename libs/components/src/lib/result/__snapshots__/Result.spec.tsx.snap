// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders correctly 1`] = `
<body>
  <div>
    <div
      data-cy="result"
    >
      <div>
        <span>
          0.00
        </span>
          USD
      </div>
      <div />
    </div>
  </div>
</body>
`;
